components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/impl/ResendCodeApiServiceImpl.java

```diff
@@ -64,7 +68,7 @@ public Response resendCodePost(ResendCodeRequestDTO resendCodeRequestDTO) {
         ResolvedUserResult resolvedUserResult = FrameworkUtils.processMultiAttributeLoginIdentification(
                 resendCodeRequestDTO.getUser().getUsername(), resendCodeRequestDTO.getUser().getTenantDomain());
         if (ResolvedUserResult.UserResolvedStatus.SUCCESS.equals(resolvedUserResult.getResolvedStatus())) {
-            resendCodeRequestDTO.getUser().setUsername(resolvedUserResult.getUser().getUsername());
+            resendCodeRequestDTO.getUser().setUsername(resolvedUserResult.getUser().getPreferredUsername());
```

Comment:
Can u check whether password recovery[1] API also affected with the same issue
[1] https://github.com/wso2-extensions/identity-governance/blob/48a9447ca343a578063b8dac7a4c6385f79e52f5/components/org.wso2.carbon.identity.api.user.recovery/src/main/java/org/wso2/carbon/identity/recovery/endpoint/impl/RecoverPasswordApiServiceImpl.java#L60

Reply:
- There is **NO** issue in password recovery API since it needs the username in the format `<USER_STORE>/<USERNAME>` for secondary user store users unlike in resend code. 

- Tested with Primary and Secondary user store users and the flows are working as expected.

Reply:
### Analysis
- For `resende-code` flow, the username is stored in the format of `<USERNAME>` in IDN_RECOVERY_DATA table but for `password-recovery` flow, the username is stored in the format of `<USER_STORE>/<USERNAME>` (only for secondary user stores) as follows.
<img width="370" alt="Screenshot 2023-12-15 at 12 42 01" src="https://github.com/wso2-extensions/identity-governance/assets/57411348/c31c2989-d786-45da-9b4e-6edc11e20ac3">

Reply:
As discussed offline let's create a git issue to track this inconsistencies

Reply:
- New git issue for this - https://github.com/wso2/product-is/issues/18628